File: apps/server/src/test/java/com/shangan/diagnostics/domain/DiagnosticLogRedactorTest.java
package com.shangan.diagnostics.domain;

import static org.assertj.core.api.Assertions.assertThat;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

/** 服务端二次脱敏：Token、口令与查询串不得进入落盘副本。 */
class DiagnosticLogRedactorTest {

  @Test
  @DisplayName("去掉 Bearer、Authorization、Cookie、JWT、口令与查询串")
  void 脱敏常见凭据() {
    String raw =
        """
        Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.aaa.bbb
        Cookie: SESSION=abc
        GET /api/v1/me?access_token=xyz
        password=hunter2
        api_key=emby-secret
        """;
    String redacted = DiagnosticLogRedactor.redact(raw);
    assertThat(redacted)
        .doesNotContain("eyJhbGciOiJIUzI1NiJ9")
        .doesNotContain("SESSION=abc")
        .doesNotContain("access_token=xyz")
        .doesNotContain("hunter2")
        .doesNotContain("emby-secret")
        .contains("[REDACTED]")
        .contains("GET /api/v1/me");
  }
}

File: apps/ios/test/core/diagnostics/diagnostic_log_redactor_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:shangan_ios/core/diagnostics/diagnostic_log_redactor.dart';

void main() {
  const redactor = DiagnosticLogRedactor();

  test('去掉 Bearer、查询串、JWT 与口令，保留路径', () {
    const raw = '''
Authorization: Bearer eyJhbGciOiJIUzI1NiJ9.aaa.bbb
GET /api/v1/me?access_token=xyz
password=hunter2
api_key=emby-secret
''';
    final out = redactor.redact(raw);
    expect(out, isNot(contains('eyJhbGciOiJIUzI1NiJ9')));
    expect(out, isNot(contains('access_token=xyz')));
    expect(out, isNot(contains('hunter2')));
    expect(out, isNot(contains('emby-secret')));
    expect(out, contains('[REDACTED]'));
    expect(out, contains('GET /api/v1/me'));
  });

  test('pathOnly 丢弃查询串', () {
    expect(
      redactor.pathOnly('https://example.test/api/v1/me?token=abc'),
      '/api/v1/me',
    );
  });
}
